src/newcharacter.cpp

```diff
@@ -1752,8 +1754,18 @@ struct {
         } else if (a == gen) {
             return true;
         }
-
-        if (sort_by_points) {
+        if( !cities_enabled ) {
+            if( a->has_flag( "CITY_START" ) == b->has_flag( "CITY_START" ) ) {
```

Comment:
You don't need to repeat all the comparisons here. You can return quickly when the two scenarios differ in the flag and otherwise continue with the existing comparison:
```C++
if( a->has_flag != b->has_flag ) {
    return a->has_flag < b->has_flag;
}
... // original code
```